apps/emqx_dashboard/src/emqx_dashboard_api.erl

```diff
@@ -33,159 +35,152 @@
-                    {message, mk(string(), #{example => "CANNOT DELETE ADMIN"})}
-                ],
-                404 => emqx_dashboard_swagger:error_codes(['USER_NOT_FOUND'], <<"User Not Found">>)
-            }
+            404 => emqx_dashboard_swagger:error_codes(['USER_NOT_FOUND'], <<"User Not Found">>)
         }
     };
-schema("/users/:username/change_pwd") ->
+
+paths("/users/:username", delete) ->
     #{
-        'operationId' => change_pwd,
-        put => #{
-            tags => [<<"dashboard">>],
-            description => <<"Update dashboard users password">>,
-            parameters => [{username, mk(binary(),
-                #{in => path, required => true, example => <<"admin">>})}],
-            'requestBody' => [
-                {old_pwd, mk(binary(), #{required => true})},
-                {new_pwd, mk(binary(), #{required => true})}
+        tags => [<<"dashboard">>],
+        description => <<"Delete dashboard users">>,
+        parameters => [{username, mk(binary(),
+            #{in => path, example => <<"admin">>})}],
+        responses => #{
+            204 => <<"Delete User successfully">>,
+            400 => [
+                {code, mk(string(), #{example => 'CANNOT_DELETE_ADMIN'})},
+                {message, mk(string(), #{example => "CANNOT DELETE ADMIN"})}
             ],
-            responses => #{
-                204 => <<"Update user password successfully">>,
-                400 => [
-                    {code, mk(string(), #{example => 'UPDATE_FAIL'})},
-                    {message, mk(string(), #{example => "Failed Reason"})}
-                ]
-            }
+            404 => emqx_dashboard_swagger:error_codes(['USER_NOT_FOUND'], <<"User Not Found">>)
         }
-    }.
+    };
 
-fields(user) ->
-    [
-        {description,
-            mk(binary(),
-                #{desc => <<"User description">>, example => "administrator"})},
-        {username,
-            mk(binary(),
-                #{desc => <<"username">>, example => "emqx"})}
-    ];
-fields(user_password) ->
-    fields(user) ++ [{password, mk(binary(), #{desc => "Password", example => <<"public">>})}].
+paths("/users/:username/change_pwd", put) ->
+    #{
+        tags => [<<"dashboard">>],
+        description => <<"Update dashboard users password">>,
+        parameters => [{username, mk(binary(),
+            #{in => path, required => true, example => <<"admin">>})}],
+        'requestBody' => [
+            {old_pwd, mk(binary(), #{required => true})},
+            {new_pwd, mk(binary(), #{required => true})}
+        ],
+        responses => #{
+            204 => <<"Update user password successfully">>,
+            400 => [
+                {code, mk(string(), #{example => 'UPDATE_FAIL'})},
+                {message, mk(string(), #{example => "Failed Reason"})}
+            ]
+        }
+    }.
 
-login(post, #{body := Params}) ->
+handle_paths("/login", post, #{body := Params}) ->
```

Comment:
I think `handle_path/3` doesn't just depend on path, it also depends on method/ body/headers.
Maybe `router` or `handle_in` or `handle`  or `handle_request` is more suitable.